### PR TITLE
performed checks before mapping

### LINKED ISSUE #3330: Take care of minorities
**Actual Behaviour**

In code there are various places where map function is implemented without checks 

**Expected Behaviour**

checks should be performed

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by anishagg17**
Fixes #3330

Changes:

performed checks before mapping

It prevents site from crashing

Demo Link: https://pr-3331-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

**Comment by anishagg17** [comment]
> Array.isArray

That is already implemented

**Comment by anishagg17** [comment]
Sir changes have been done!

File: src/components/About/Blog/index.js [hunk 1]
```diff
@@ -305,140 +305,144 @@ class Blog extends Component {
...
-                let category = [];
-                posts.categories.forEach(cat => {
-                  let k = 0;
-                  for (k = 0; k < allCategories.length; k++) {
-                    if (cat === allCategories[k]) {
-                      category.push(cat);
+              {this.state.posts &&
+                this.state.posts.length > 0 &&
+                Array.isArray(this.state.posts) &&
+                this.state.posts.map((posts, i) => {
```
Comment (akshatnitd) [inline src/components/About/Blog/index.js]: If `this.state.posts` isn't an array, then `this.state.posts.length ` will fail and break the code. Add isArray check before checking length.

Comment (anishagg17) [reply src/components/About/Blog/index.js]: OKay

Comment (anishagg17) [reply src/components/About/Blog/index.js]: Done Sir!

**Pull request merged**